### ISSUE TITLE
install-node.sh script incorrectly detects Teleport is already running

### ISSUE THREAD

**Issue by lbogdan** [gravitational/teleport#34226]
Scenario:
- a Proxmox host;
- an LXC container running on said Proxmox host.

I want to add both to Teleport, first the LXC container - this works, then the Proxmox host - this fails.

**Expected behavior:**

Adding the Proxmox host works.

**Current behavior:**

The `install-node.sh` script fails on the Proxmox host with the following error:

```sh
sudo bash -c "$(curl -fsSL https://[redacted]/scripts/[redacted]/install-node.sh)"
# 2023-11-04 21:31:00 UTC [teleport-installer] TELEPORT_VERSION: 14.1.1
# 2023-11-04 21:31:00 UTC [teleport-installer] TARGET_HOSTNAME: [redacted]
# 2023-11-04 21:31:00 UTC [teleport-installer] TARGET_PORT: 443
# 2023-11-04 21:31:00 UTC [teleport-installer] JOIN_TOKEN: [redacted]
# 2023-11-04 21:31:00 UTC [teleport-installer] CA_PIN_HASHES: sha256:[redacted]
# 2023-11-04 21:31:00 UTC [teleport-installer] Checking TCP connectivity to Teleport server ([redacted]:443)
# 2023-11-04 21:31:00 UTC [teleport-installer] Connectivity to Teleport server (via nc) looks good
# 2023-11-04 21:31:00 UTC [teleport-installer] Detected host: linux-gnu, using Teleport binary type linux
# 2023-11-04 21:31:00 UTC [teleport-installer] Detected arch: x86_64, using Teleport arch amd64
# 2023-11-04 21:31:00 UTC [teleport-installer] Detected distro type: debian
# 2023-11-04 21:31:00 UTC [teleport-installer] Using Teleport distribution: deb
# 2023-11-04 21:31:00 UTC [teleport-installer] Created temp dir /tmp/teleport-[redacted]
#
# Warning: Teleport appears to already be running on this host (pid: [redacted])
#
# [...]
# Run this installer again when done.
```

This is because the LXC container's Teleport process also shows on the host, its PID is the one showing in the script output above.

The workaround was to download the script and edit it to `return 1` early in `is_running_teleport()`, which then worked fine.

https://github.com/gravitational/teleport/blob/a717e8f7a04fd523ed6166bcf8677b944c0982db/lib/web/scripts/node-join/install.sh#L497-L501

I'm not sure how this can be detected reliably, in my particular case the grand-parent of the `teleport` process is `lxc-start`:

```
/usr/bin/lxc-start -F -n [redacted]
 \_ /sbin/init
 [...]
      \_ /usr/local/bin/teleport start --config /etc/teleport.yaml --pid-file=/run/teleport.pid
```

If you think that's a reasonable approach, I can open a PR.

**Bug details:**
- Teleport version: 14.1.1
- Recreation steps: see above
- Debug logs: see above

**Comment by lbogdan** [comment]
Going through the `install-node.sh` code, I see there's an `-i` option to ignore the "Teleport is already running" check, but it's not at all discoverable, as it doesn't even show in the usage:

```
bash install-node.sh -h
# Usage: install-node.sh [-v teleport_version] [-h target_hostname] [-p target_port] [-j join_token] [-c ca_pin_hash]... [-q] [-l log_filename] [-a app_name] [-u app_uri]
```

**Comment by zmb3** [comment]
I don't think we want to get into the rabbit hole that is checking parent process and looking specifically for `lxc-start`, especially given that this is probably more of a niche use case.

If I were you, I would use the installer script once (probably directly on the host), and then join the LXC container manually (by creating a token with `tctl tokens add`).

**Comment by lbogdan** [comment]
Agreed it's a rabbit hole, and I wouldn't have hit it if I first added the Proxmox machine, and then the LXC container.

Also, `-i` works around it, but you'd have to read the source to figure out there's even such an option.

I'll close this for now, hopefully people hitting the issue will find this helpful.

**Comment by webvictim** [comment]
This should have been fixed now by https://github.com/gravitational/teleport/pull/59706

Not released yet, but should be out with v18.3.0.
